{ip,nf}tablesdoc: ensure format compatibility across Debian versions

akerouanton opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

- Related to https://github.com/moby/moby/pull/50726

**- What I did**

**iptablesdoc: remove -n from iptables -L invocations**

The output of `iptables -nvL` has changed in Debian 13 — the proto column now shows protocol names instead of numbers, even when `-n` is specified. This breaks the iptablesdoc golden files, which expect protocols to be represented numerically.

This change comes from 34f085b1607364f4eaded1140060dcaf965a2649 in repo git.netfilter.org/iptables (see [1]), which is a revert of da8ecc62dd765b15df84c3aa6b83dcb7a81d4ffa (see [2]), and was made to address a bug report (see [3]).

Unfortunately, this means there's a drift between iptables versions. So, remove the `-n` flag altogether to ensure that the iptablesdoc tests pass everywhere.

[1]: https://git.netfilter.org/iptables/commit/?id=da8ecc62dd765b15df84c3aa6b83dcb7a81d4ffa
[2]: https://git.netfilter.org/iptables/commit/?id=34f085b1607364f4eaded1140060dcaf965a2649
[3]: https://bugzilla.netfilter.org/show_bug.cgi?id=1729

**nftablesdoc: stringify numerical dstnat prio**

When nftablesdoc tests dump the state of nftables, the argument '-y' / '--numeric-priority' isn't used, so all priorities should be stringified. However, there's a bug in older versions of nftables that prevents the stringification of the 'dstnat' priority — it's currently dumped as '-100'.

New versions fix that, and thus running these tests on Debian 13 fails because of this discrepancy with golden files.

So, look for 'type nat hook output priority -100' and stringify the priority to ensure compatibility across versions of nft.

akerouanton committed:
iptablesdoc: remove -n from iptables -L invocations

The output of `iptables -nvL` has changed in Debian 13 — the proto
column now shows protocol names instead of numbers, even when `-n` is
specified. This breaks the iptablesdoc golden files, which expect
protocols to be represented numerically.

This change comes from 34f085b1607364f4eaded1140060dcaf965a2649 in repo
git.netfilter.org/iptables (see [1]), which is a revert of
da8ecc62dd765b15df84c3aa6b83dcb7a81d4ffa (see [2]), and was made to
address a bug report (see [3]).

Unfortunately, this means there's a drift between iptables versions. So,
remove the `-n` flag altogether to ensure that the iptablesdoc tests
pass everywhere.

[1]: https://git.netfilter.org/iptables/commit/?id=da8ecc62dd765b15df84c3aa6b83dcb7a81d4ffa
[2]: https://git.netfilter.org/iptables/commit/?id=34f085b1607364f4eaded1140060dcaf965a2649
[3]: https://bugzilla.netfilter.org/show_bug.cgi?id=1729

Signed-off-by: Albin Kerouanton <albinker@gmail.com>

thaJeztah commented:
From the other PR; looks like there's also a tests in `daemon/libnetwork/drivers/bridge/internal/nftabler` that fail;

```
=== FAIL: daemon/libnetwork/drivers/bridge/internal/nftabler TestNftabler/ipv4=true/ipv6=false/hairpin=false/internal=false/icc=false/masq=true/snat=true/gwm=routed/bindlh=false/wsl2mirrored=false (0.03s)
    nftabler_test.go:130: assertion failed: 
        --- expected
        +++ actual
        @@ -23,5 +23,5 @@
         
         	chain nat-OUTPUT {
        -		type nat hook output priority -100; policy accept;
        +		type nat hook output priority dstnat; policy accept;
         		ip daddr != 127.0.0.0/8 fib daddr type local counter packets 0 bytes 0 jump nat-prerouting-and-output
         	}
        
        
        You can run 'go test . -update' to automatically update testdata/TestNftabler_cleaned,hairpin=false__ip.golden to the new expected value.'
```

thaJeztah reviewed (approved):
LGTM


not merging yet, as it'd probably be good to have the other changes in the same PR to keep it all together (but can be separate commits).

akerouanton committed:
nftabler,nftablesdoc: stringify numerical dstnat prio

When nftablesdoc tests dump the state of nftables, the argument '-y' /
'--numeric-priority' isn't used, so all priorities should be
stringified. However, there's a bug in older versions of nftables that
prevents the stringification of the 'dstnat' priority — it's currently
dumped as '-100'.

New versions fix that, and thus running these tests on Debian 13 fails
because of this discrepancy with golden files.

So, look for 'type nat hook output priority -100' and stringify the
priority to ensure compatibility across versions of nft.

Signed-off-by: Albin Kerouanton <albinker@gmail.com>

thaJeztah commented on daemon/libnetwork/drivers/bridge/internal/nftabler/testdata/TestNftabler_cleaned,hairpin=false,wsl2mirrored=true__ip.golden:
Not a blocker; but should we take the opportunity to move these to a `TestNftabler` subdirectory, as I did in that other PR?

- https://github.com/moby/moby/pull/50747

akerouanton commented on integration/network/bridge/iptablesdoc/generated/usernet-portmap-routed.md:
```diff
@@ -108,10 +108,10 @@ for `prot 1`:
-    4        0     0 DROP       0    --  !bridge1 bridge1  0.0.0.0/0            0.0.0.0/0           
+    1        0     0 ACCEPT     tcp  --  !bridge1 bridge1  anywhere             192.0.2.2            tcp dpt:http
+    2        0     0 DROP       all  --  !docker0 docker0  anywhere             anywhere            
+    3        0     0 ACCEPT     icmp --  any    bridge1  anywhere             anywhere            
```
Good catch! Fixed in the last force-push.
